oidc-api/src/main/java/uk/gov/di/authentication/oidc/services/AccountInterventionService.java

```diff
@@ -0,0 +1,77 @@
+package uk.gov.di.authentication.oidc.services;
+
+import com.google.gson.Gson;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import uk.gov.di.authentication.oidc.entity.AccountInterventionResponse;
+import uk.gov.di.authentication.oidc.entity.AccountInterventionStatus;
+import uk.gov.di.authentication.oidc.exceptions.AccountInterventionException;
+import uk.gov.di.orchestration.shared.services.AuditService;
+import uk.gov.di.orchestration.shared.services.ConfigurationService;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+
+public class AccountInterventionService {
+
+    private static final Logger LOGGER = LogManager.getLogger(AccountInterventionService.class);
+    private final boolean accountInterventionsAuditEnabled;
+    private final boolean accountInterventionsEnabled;
+    private final HttpClient httpClient;
+    private final URI accountInterventionServiceURI;
+    private final AuditService auditService;
+
+    public AccountInterventionService(
+            ConfigurationService configService, AuditService auditService, HttpClient httpClient) {
+        this.accountInterventionsAuditEnabled =
+                configService.isAccountInterventionServiceAuditEnabled();
+        this.accountInterventionsEnabled = configService.isAccountInterventionServiceEnabled();
+        this.accountInterventionServiceURI = configService.getAccountInterventionServiceURI();
+        this.auditService = auditService;
+        this.httpClient = httpClient;
+    }
+
+    public AccountInterventionStatus getAccountStatus(String internalSubjectId)
+            throws AccountInterventionException {
+        try {
+            if (accountInterventionsEnabled) {
+                return retrieveAccountStatus(internalSubjectId);
+            }
+
+            if (accountInterventionsAuditEnabled) {
+                sendAuditEvent();
+            }
```

Comment:
Surely we won't return send the audit event here if we get a valid response, because we'll have already returned the status?

Reply:
Yes, of course. Maybe I really should have ignored the audit entirely for now.